### ISSUE TITLE
[Dashboard] Service mesh dashboard not displaying provisioned Istio mesh

### ISSUE THREAD

**Issue by leecalcote** [meshery/meshery#6367]
#### Current Behavior
✅ MeshSync running
✅ Meshery Broker connected and Istio discovered. 

<img width="2549" alt="Screen Shot 2022-10-10 at 2 29 10 PM" src="https://user-images.githubusercontent.com/7570704/194939376-0dd07ce4-06f2-4316-bd2c-d34df0a3951c.png">

#### Environment

- **Host OS:** Mac
- **Meshery Server Version:** stable-v0.6.15
- **Meshery Client Version:** stable-v0.6.15
- **Platform:**  Kubernetes v1.25

```
$ k version --short
Flag --short has been deprecated, and will be removed in the future. The --short output will become the default.
Client Version: v1.25.2
Kustomize Version: v4.5.7
Server Version: v1.25.0
```

<img width="1277" alt="Screen Shot 2022-10-10 at 2 33 47 PM" src="https://user-images.githubusercontent.com/7570704/194939842-38fdfc3c-502d-480c-90ed-f9bb81a5f483.png">

##### Database

[database.zip](https://github.com/meshery/meshery/files/9748493/config.zip)

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by leecalcote** [comment]
I'm seeing new error logs now:

```
rt-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint; ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
ERRO[2022-10-10T17:00:51-05:00] ON CONFLICT clause does not match any PRIMARY KEY or UNIQUE constraint  app=meshery code=2102 probable-cause= severity=2 short-description="Unable to Persist config data." suggested-remediation=
```

**Comment by leecalcote** [comment]
After a database reset, I'm now seeing the service mesh dashboard populated.

**Comment by leecalcote** [comment]
Lowering priority of this bug.

**Comment by theBeginner86** [comment]
I saw this issue. It is not a bug. 
Since I updated the database schema so existing databases with older schema would not allow data to be stored as now the latest meshery release would refer to a new struct in MeshSync with different constraint values (added through gorm annotation).
This is just a one-time experience for users with an existing database carved with older MeshSync structs. Once they purge their DB then existing tables would be truncated and new tables would be created with new constraints and things would again function as they used to.

Though new users would not face this as the latest release would always refer to new structs and create fresh DB.
